client/my-sites/plan-features-2023-grid/_media-queries.scss

```diff
@@ -106,6 +106,13 @@ $planComparisonMinWidthToGridWidthWithSidebarMap: (
 			display: none;
 		}
 	}
+
+	.plan-features-2023-grid__current-plan {
+		display: none;
+		@media ( min-width: $plans-2023-small-breakpoint ) {
+			display: block;
+		}
```

Comment:
I still can't fathom how we ended up with a design like this (render everything and hide via CSS), and we keep adding to it. This will need to be refactored next.

How big of a deal would it be to move the logic to JS, for this case alone, and pass a variable through from `PlansFeaturesMain` whether to highlight a plan or not (and pass the plan-slug/type to highlight)? Can you think of a minimal solution to just get this case done that way?

Reply:
> I still can't fathom how we ended up with a design like this (render everything and hide via CSS), and we keep adding to it. This will need to be refactored next.

100% agree!

> How big of a deal would it be to move the logic to JS, for this case alone, and pass a variable through from PlansFeaturesMain whether to highlight a plan or not (and pass the plan-slug/type to highlight)? Can you think of a minimal solution to just get this case done that way?

I feel quite strongly about doing this as a separate change which would be easier to review and test instead of combining it with this. If we can add JS logic just for this case, it will be fairly trivial to show/hide the desktop/tablet/mobile views as well. At the top of my mind, this would be a `useGetScreenType` hook which would observe the screen width and return `mobile | tablet | desktop`. After that we could do something like this:
```js
{ screenType === 'desktop' && (
	<div className="plan-features-2023-grid__desktop-view">
		{ this.renderTable( planProperties ) }
	</div>
) }
{ screenType === 'tablet' && (
	<div className="plan-features-2023-grid__tablet-view">
		{ this.renderTabletView() }
	</div>
) }
{ screenType === 'mobile' && (
	<div className="plan-features-2023-grid__mobile-view">
		{ this.renderMobileView( planPropertiesForMobileView ) }
	</div>
) }
```

I have created an issue for tracking this here: https://github.com/Automattic/martech/issues/1912

Reply:
I'm not too sure about this. It seems to just let the root problem live on and pass the principle of complexity to the consuming end (to determine which to show when).

So I would wish to see us simplify this to two possible views: `mobile`, and everything else. that's pretty much how it works now, with the "tablet" and "desktop" views being just a difference in number of columns. It's the implementation that suffers, with these over-engineered "screen types", media queries, different structures used for mobile, and everything else that stems from it. 

The first thing I'd consider is if a `min-width` on the columns (and some programmatic computation around it) would suffice to adapt the desktop view until the mobile (card-based) one sets in. I'd consider passing an optional `isMobile` prop to force the card-based/mobile view on certain viewports (which should cover the Calypso side with the variable sidebar, etc.). On default, the mobile view should set in/out after a certain query change.

For this case (this change), it would help to pass the slug of the plan to highlight given a certain viewport change (and avoid adding more to these media queries). Whatever we create for this will most likely help with what I described above. I think every chance we get right now to do a little bit of refactoring, we should take the opportunity.

^ @southp

Reply:
> So I would wish to see us simplify this to two possible views: mobile, and everything else. 

In that case, we should go for a single flex-based layout. After all, the expanded mobile card is almost exactly the same as the columns on desktop/tablet. We have fixed values for grid width and the number of columns corresponding to the screen width, and those would have to be implemented using media queries. We can continue the discussion in https://github.com/Automattic/wp-calypso/issues/79010, since this is beyond the scope of this PR.

> For this case (this change), it would help to pass the slug of the plan to highlight given a certain viewport change (and avoid adding more to these media queries). 

Maybe I misunderstood this - reacting to a viewport change would still need a media query, right? Can you please explain, with a pseudo-code example, if possible?

Reply:
> In that case, we should go for a single flex-based layout. After all, the expanded mobile card is almost exactly the same as the columns on desktop/tablet. We have fixed values for grid width and the number of columns corresponding to the screen width, and those would have to be implemented using media queries. We can continue the discussion in https://github.com/Automattic/wp-calypso/issues/79010, since this is beyond the scope of this PR.

I'm not sure if rewriting the grid implementation is the best use of time to reach the package/NPM milestone. A grid or flex design should have been the consideration from the getgo. Right now, we should be unblocking this while maintaining a decent API over it (at the cost of a bit of extra processing). An internal refactor can happen later if you are up for it.

> reacting to a viewport change would still need a media query, right

We have viewport / viewport-react packages that you can use. I think the point of investigation would be to determine if sidebar is opened or not. From there, you can add a layout effect and pass the right prop through.

Reply:
> I think the point of investigation would be to determine if sidebar is opened or not.

We do have a `getSidebarIsCollapsed` selector that gives us this info.

> From there, you can add a layout effect and pass the right prop through.

An additional challenge here is that in mobile view, the current plan should be at the top of the list. Due the way `plans` and `visiblePlans` are structured and how `planProperties` are calculated, it adds additional complexity to this change. It would all be easier if we move all the plan properties computation to PlanFeaturesMain in this case.

Reply:
> It would all be easier if we move all the plan properties computation to PlanFeaturesMain in this case.

These are happening in other PRs already. A couple of hefty refactors will show up soon. I'm not sure it makes a difference though.

For what you are working on, the exploration suggested above would help to also contribute overall, as that's functionality we will need. So avoiding the media query would lead the way for others. 

What is the minimum implementation needed to pass a "spotlight" plan slug (that can be current plan, but also any slug) through from PlansFeaturesMain, on certain viewport, and also have it filtered out of visible plans (also prior to sending through as prop)? Mobile view should be getting none and the full set of visible plans. Is this direction blocked due to something?